--- origsrc/gnome-settings-daemon-2.22.2.1/configure.ac	1974-05-19 21:39:46.000000000 -0500
+++ src/gnome-settings-daemon-2.22.2.1/configure.ac	2008-06-04 00:38:23.515625000 -0500
@@ -75,8 +75,17 @@
 PKG_CHECK_MODULES(LIBSOUNDS, gio-2.0 libgnome-2.0 libgnomeui-2.0)
 
 
-GSD_PLUGIN_LDFLAGS="-export_dynamic -module -avoid-version -no-undefined"
+GSD_PLUGIN_LDFLAGS="-export_dynamic -module -avoid-version"
+case "$host_os" in
+	cygwin*)
+		platform_win32=yes
+		GSD_PLUGIN_LDFLAGS="$GSD_PLUGIN_LDFLAGS -no-undefined -Wl,\$(top_builddir)/gnome-settings-daemon/libgsd.a"
+		;;
+	*)	platform_win32=no
+		;;
+esac
 AC_SUBST([GSD_PLUGIN_LDFLAGS])
+AM_CONDITIONAL(PLATFORM_WIN32, test "$platform_win32" = "yes")
 
 AC_PATH_PROG(GLIB_GENMARSHAL, glib-genmarshal)
 AC_PATH_XTRA
--- origsrc/gnome-settings-daemon-2.22.2.1/gnome-settings-daemon/Makefile.am	1974-03-20 19:48:41.000000000 -0600
+++ src/gnome-settings-daemon-2.22.2.1/gnome-settings-daemon/Makefile.am	2008-06-04 00:40:53.671875000 -0500
@@ -74,6 +74,11 @@
 gnome_settings_daemon_LDFLAGS = 	\
 	$(AM_LDFLAGS)
 
+if PLATFORM_WIN32
+gnome_settings_daemon_LDFLAGS += 	\
+	-Wl,--export-all-symbols -Wl,--out-implib,libgsd.a
+endif
+
 gnome_settings_daemon_LDADD = 		\
 	libgsd-profile.la		\
 	$(SETTINGS_DAEMON_LIBS)
