--- origsrc/gnome-settings-daemon-2.26.0/configure.ac	2009-03-16 09:52:06.000000000 -0500
+++ src/gnome-settings-daemon-2.26.0/configure.ac	2009-04-03 01:08:13.290701700 -0500
@@ -54,6 +54,10 @@ LIBNOTIFY_REQUIRED_VERSION=0.4.3
 
 EXTRA_COMPILE_WARNINGS(yes)
 
+PKG_CHECK_MODULES(LIBGSD_PLUGINS,
+        gobject-2.0 >= $GLIB_REQUIRED_VERSION
+)
+
 PKG_CHECK_MODULES(SETTINGS_DAEMON,
         glib-2.0 >= $GLIB_REQUIRED_VERSION
         gtk+-2.0 >= $GTK_REQUIRED_VERSION
--- origsrc/gnome-settings-daemon-2.26.0/gnome-settings-daemon/Makefile.am	2009-03-16 09:35:39.000000000 -0500
+++ src/gnome-settings-daemon-2.26.0/gnome-settings-daemon/Makefile.am	2009-04-03 01:09:23.150076700 -0500
@@ -10,7 +10,7 @@ INCLUDES = \
 	$(SETTINGS_DAEMON_CFLAGS)				\
 	$(NULL)
 
-noinst_LTLIBRARIES = 			\
+lib_LTLIBRARIES = 			\
 	libgsd-plugins.la		\
 	$(NULL)
 
@@ -29,10 +29,12 @@ libgsd_plugins_la_CFLAGS =		\
 	$(NULL)
 
 libgsd_plugins_la_LIBADD =		\
+	$(LIBGSD_PLUGINS_LIBS)		\
 	$(NULL)
 
 libgsd_plugins_la_LDFLAGS = 		\
 	-export-dynamic 		\
+	-no-undefined			\
 	$(NULL)
 
 gsddir = $(libexecdir)
