--- origsrc/gnome-settings-daemon-2.28.1/configure.ac	2009-10-19 09:04:07.000000000 -0500
+++ src/gnome-settings-daemon-2.28.1/configure.ac	2009-11-01 03:28:50.638416800 -0600
@@ -54,6 +54,10 @@ LIBNOTIFY_REQUIRED_VERSION=0.4.3
 
 EXTRA_COMPILE_WARNINGS(yes)
 
+PKG_CHECK_MODULES(GOBJECT,
+        gobject-2.0 >= $GLIB_REQUIRED_VERSION
+)
+
 PKG_CHECK_MODULES(SETTINGS_DAEMON,
         glib-2.0 >= $GLIB_REQUIRED_VERSION
         gtk+-2.0 >= $GTK_REQUIRED_VERSION
@@ -70,7 +74,7 @@ PKG_CHECK_MODULES(SETTINGS_PLUGIN,
         dbus-glib-1 >= $DBUS_GLIB_REQUIRED_VERSION
 )
 
-GSD_PLUGIN_LDFLAGS="-export_dynamic -module -avoid-version -no-undefined"
+GSD_PLUGIN_LDFLAGS="-export-dynamic -module -avoid-version -no-undefined \$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la"
 case $host_os in
   darwin*)
     GSD_PLUGIN_LDFLAGS="${GSD_PLUGIN_LDFLAGS} -Wl,-bundle_loader,\$(top_builddir)/gnome-settings-daemon/gnome-settings-daemon"
--- origsrc/gnome-settings-daemon-2.28.1/gnome-settings-daemon/Makefile.am	2009-05-03 13:15:30.000000000 -0500
+++ src/gnome-settings-daemon-2.28.1/gnome-settings-daemon/Makefile.am	2009-11-01 03:28:05.585538300 -0600
@@ -10,28 +10,32 @@ INCLUDES = \
 	$(SETTINGS_DAEMON_CFLAGS)				\
 	$(NULL)
 
-noinst_LTLIBRARIES = 			\
-	libgsd-profile.la		\
+lib_LTLIBRARIES = 			\
+	libgsd-plugins.la		\
 	$(NULL)
 
-libgsd_profile_la_SOURCES =		\
+libgsd_plugins_la_SOURCES =		\
+	gnome-settings-plugin.c		\
+	gnome-settings-plugin.h		\
 	gnome-settings-profile.c	\
 	gnome-settings-profile.h	\
 	$(NULL)
 
-libgsd_profile_la_CPPFLAGS = 		\
+libgsd_plugins_la_CPPFLAGS = 		\
 	$(DISABLE_DEPRECATED_CFLAGS)	\
 	$(NULL)
 
-libgsd_profile_la_CFLAGS =		\
-	$(SIMPLE_GREETER_CFLAGS)	\
+libgsd_plugins_la_CFLAGS =		\
+	$(GOBJECT_CFLAGS)	\
 	$(NULL)
 
-libgsd_profile_la_LIBADD =		\
+libgsd_plugins_la_LIBADD =		\
+	$(GOBJECT_LIBS)	\
 	$(NULL)
 
-libgsd_profile_la_LDFLAGS = 	\
+libgsd_plugins_la_LDFLAGS = 	\
 	-export-dynamic 	\
+	-no-undefined		\
 	$(NULL)
 
 gsddir = $(libexecdir)
@@ -59,8 +63,6 @@ gnome_settings_daemon_SOURCES =		\
 	main.c				\
 	gnome-settings-manager.c	\
 	gnome-settings-manager.h	\
-	gnome-settings-plugin.c		\
-	gnome-settings-plugin.h		\
 	gnome-settings-plugin-info.c	\
 	gnome-settings-plugin-info.h	\
 	gnome-settings-module.c		\
@@ -77,7 +79,7 @@ gnome_settings_daemon_LDFLAGS = 	\
 	$(AM_LDFLAGS)
 
 gnome_settings_daemon_LDADD = 		\
-	libgsd-profile.la		\
+	libgsd-plugins.la		\
 	$(SETTINGS_DAEMON_LIBS)
 
 EXTRA_DIST = 				\
--- origsrc/gnome-settings-daemon-2.28.1/plugins/background/Makefile.am	2009-05-03 13:15:30.000000000 -0500
+++ src/gnome-settings-daemon-2.28.1/plugins/background/Makefile.am	2009-11-01 11:37:02.294095600 -0600
@@ -20,7 +20,7 @@ test_background_CFLAGS = \
 	$(AM_CFLAGS)
 
 test_background_LDADD =		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-profile.la	\
+	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(NULL)
 
--- origsrc/gnome-settings-daemon-2.28.1/plugins/media-keys/Makefile.am	2009-09-21 05:24:02.000000000 -0500
+++ src/gnome-settings-daemon-2.28.1/plugins/media-keys/Makefile.am	2009-11-01 11:37:26.910938500 -0600
@@ -117,7 +117,7 @@ test_media_keys_CFLAGS = \
 	$(AM_CFLAGS)
 
 test_media_keys_LDADD = \
-	$(top_builddir)/gnome-settings-daemon/libgsd-profile.la		\
+	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la		\
 	$(top_builddir)/plugins/common/libcommon.la			\
 	$(SETTINGS_DAEMON_LIBS)			\
 	$(SETTINGS_PLUGIN_LIBS)			\
