--- origsrc/gnome-settings-daemon-3.6.3/configure.ac	2012-12-19 16:55:41.314691200 -0600
+++ src/gnome-settings-daemon-3.6.3/configure.ac	2012-12-19 17:08:10.622549200 -0600
@@ -241,12 +239,6 @@ if test x$have_gudev != xno; then
 fi
 
 dnl ---------------------------------------------------------------------------
-dnl - color
-dnl ---------------------------------------------------------------------------
-
-PKG_CHECK_MODULES(COLOR, [colord >= 1.0.2 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3])
-
-dnl ---------------------------------------------------------------------------
 dnl - datetime
 dnl ---------------------------------------------------------------------------
 
@@ -507,3 +499,2 @@ plugins/a11y-settings/Makefile
 plugins/clipboard/Makefile
-plugins/color/Makefile
 plugins/common/Makefile
--- origsrc/gnome-settings-daemon-3.6.3/plugins/Makefile.am	2012-12-19 16:55:43.013788400 -0600
+++ src/gnome-settings-daemon-3.6.3/plugins/Makefile.am	2012-12-19 17:09:52.739390000 -0600
@@ -6,3 +6,2 @@ enabled_plugins =	\
 	clipboard	\
-	color		\
 	cursor		\
