--- origsrc/gnome-settings-daemon-3.14.2/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in	2013-03-24 17:32:38.000000000 -0500
+++ src/gnome-settings-daemon-3.14.2/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in	2015-01-18 01:43:18.254940300 -0600
@@ -36,7 +36,7 @@
       <_description>A list of strings representing the GTK+ modules that will be loaded, usually in addition to conditional and forcibly disabled ones.</_description>
     </key>
     <key type="a{sv}" name="overrides">
-      <default>{}</default>
+      <default><![CDATA[{'Gtk/ShellShowsAppMenu': <0>}]]></default>
       <summary>A dictionary of XSETTINGS to override</summary>
       <description>This dictionary maps XSETTINGS names to overrides values. The values must be either strings, signed int32s or (in the case of colors), 4-tuples of uint16 (red, green, blue, alpha; 65535 is fully opaque).</description>
     </key>
