--- origsrc/gnome-settings-daemon-3.14.2/configure.ac	2015-01-18 03:30:47.987844000 -0600
+++ src/gnome-settings-daemon-3.14.2/configure.ac	2015-01-18 03:30:48.975900500 -0600
@@ -205,7 +205,7 @@ dnl ------------------------------------
 dnl - media-keys plugin stuff
 dnl ---------------------------------------------------------------------------
 
-PKG_CHECK_MODULES(MEDIA_KEYS, [gio-unix-2.0 libpulse >= $PA_REQUIRED_VERSION $GUDEV_PKG libpulse-mainloop-glib >= $PA_REQUIRED_VERSION libcanberra-gtk3 upower-glib >= $UPOWER_REQUIRED_VERSION])
+PKG_CHECK_MODULES(MEDIA_KEYS, [gio-unix-2.0 libpulse >= $PA_REQUIRED_VERSION $GUDEV_PKG libpulse-mainloop-glib >= $PA_REQUIRED_VERSION libcanberra-gtk3])
 PKG_CHECK_MODULES(GVC, [gobject-2.0 libpulse >= $PA_REQUIRED_VERSION libpulse-mainloop-glib >= $PA_REQUIRED_VERSION])
 AM_CONDITIONAL(HAVE_INTROSPECTION, false)
 
@@ -213,7 +213,7 @@ dnl ------------------------------------
 dnl - xrandr plugin stuff
 dnl ---------------------------------------------------------------------------
 
-PKG_CHECK_MODULES(XRANDR, [gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION upower-glib >= $UPOWER_REQUIRED_VERSION])
+PKG_CHECK_MODULES(XRANDR, [gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION])
 
 dnl ---------------------------------------------------------------------------
 dnl - orientation plugin stuff
@@ -229,44 +229,6 @@ dnl ------------------------------------
 
 PKG_CHECK_MODULES(SOUND, [libpulse >= $PA_REQUIRED_VERSION $GUDEV_PKG libpulse-mainloop-glib >= $PA_REQUIRED_VERSION])
 
-# ---------------------------------------------------------------------------
-# Power
-# ---------------------------------------------------------------------------
-PKG_CHECK_MODULES(POWER, upower-glib >= $UPOWER_REQUIRED_VERSION gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION $GUDEV_PKG libcanberra-gtk3 libnotify x11 xext xtst)
-
-if test x$have_gudev != xno; then
-	PKG_CHECK_MODULES(BACKLIGHT_HELPER,
-		glib-2.0 >= $GLIB_REQUIRED_VERSION
-		gudev-1.0
-	)
-fi
-
-dnl ---------------------------------------------------------------------------
-dnl - color
-dnl ---------------------------------------------------------------------------
-
-PKG_CHECK_MODULES(COLOR, [colord >= 1.0.2 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3])
-
-dnl ---------------------------------------------------------------------------
-dnl - datetime
-dnl ---------------------------------------------------------------------------
-
-AC_CHECK_LIBM
-AC_SUBST(LIBM)
-
-PKG_CHECK_MODULES(DATETIME,
-        geoclue-2.0 >= $GEOCLUE_REQUIRED_VERSION
-	geocode-glib-1.0 >= $GEOCODE_GLIB_REQUIRED_VERSION
-	gweather-3.0 >= $LIBGWEATHER_REQUIRED_VERSION
-	polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION
-)
-
-GEOCLUE_DBUS_INTERFACE_XML=`pkg-config --variable=dbus_interface geoclue-2.0`
-if test "x$GEOCLUE_DBUS_INTERFACE_XML" = "x"; then
-	AC_MSG_ERROR([Cannot find dbus_interface variable in geoclue-2.0.pc])
-fi
-AC_SUBST(GEOCLUE_DBUS_INTERFACE_XML)
-
 dnl ---------------------------------------------------------------------------
 dnl - wacom (disabled for s390/s390x and non Linux platforms)
 dnl ---------------------------------------------------------------------------
--- origsrc/gnome-settings-daemon-3.14.2/plugins/Makefile.am	2014-09-22 11:02:18.000000000 -0500
+++ src/gnome-settings-daemon-3.14.2/plugins/Makefile.am	2015-01-18 03:30:48.978900600 -0600
@@ -4,11 +4,8 @@ enabled_plugins =	\
 	a11y-keyboard	\
 	a11y-settings	\
 	clipboard	\
-	color		\
 	cursor		\
-	datetime	\
 	dummy		\
-	power		\
 	housekeeping    \
 	keyboard	\
 	media-keys	\
--- origsrc/gnome-settings-daemon-3.14.2/plugins/media-keys/gsd-media-keys-manager.c	2015-01-18 03:30:47.806833600 -0600
+++ src/gnome-settings-daemon-3.14.2/plugins/media-keys/gsd-media-keys-manager.c	2015-01-18 03:30:48.983900900 -0600
@@ -39,8 +39,6 @@
 #include <gio/gdesktopappinfo.h>
 #include <gio/gunixfdlist.h>
 
-#include <libupower-glib/upower.h>
-
 #ifdef HAVE_GUDEV
 #include <gudev/gudev.h>
 #endif
@@ -97,7 +95,6 @@ static const gchar introspection_xml[] =
 "</node>";
 
 #define SETTINGS_INTERFACE_DIR "org.gnome.desktop.interface"
-#define SETTINGS_POWER_DIR "org.gnome.settings-daemon.plugins.power"
 #define SETTINGS_XSETTINGS_DIR "org.gnome.settings-daemon.plugins.xsettings"
 #define SETTINGS_TOUCHPAD_DIR "org.gnome.settings-daemon.peripherals.touchpad"
 #define TOUCHPAD_ENABLED_KEY "touchpad-enabled"
@@ -157,13 +154,6 @@ struct GsdMediaKeysManagerPrivate
         char            *icon_theme;
         char            *gtk_theme;
 
-        /* Power stuff */
-        GSettings       *power_settings;
-        GDBusProxy      *power_proxy;
-        GDBusProxy      *power_screen_proxy;
-        GDBusProxy      *power_keyboard_proxy;
-        UpDevice        *composite_device;
-
         /* Shell stuff */
         GsdShell        *shell_proxy;
         ShellKeyGrabber *key_grabber;
@@ -1779,6 +1769,7 @@ do_toggle_contrast_action (GsdMediaKeysM
 	}
 }
 
+#if 0
 static void
 power_action (GsdMediaKeysManager *manager,
               const char          *action,
@@ -1935,6 +1926,7 @@ do_battery_action (GsdMediaKeysManager *
 
         g_free (icon_name);
 }
+#endif
 
 static void
 screencast_stop (GsdMediaKeysManager *manager)
@@ -2138,38 +2130,19 @@ do_action (GsdMediaKeysManager *manager,
 		do_toggle_contrast_action (manager);
 		break;
         case POWER_KEY:
-                do_config_power_action (manager, "button-power", FALSE);
-                break;
         case SLEEP_KEY:
-                do_config_power_action (manager, "button-sleep", FALSE);
-                break;
         case SUSPEND_KEY:
-                do_config_power_action (manager, "button-suspend", FALSE);
-                break;
         case HIBERNATE_KEY:
-                do_config_power_action (manager, "button-hibernate", FALSE);
-                break;
         case POWER_KEY_NO_DIALOG:
-                do_config_power_action (manager, "button-power", TRUE);
-                break;
         case SLEEP_KEY_NO_DIALOG:
-                do_config_power_action (manager, "button-sleep", TRUE);
-                break;
         case SUSPEND_KEY_NO_DIALOG:
-                do_config_power_action (manager, "button-suspend", TRUE);
-                break;
         case HIBERNATE_KEY_NO_DIALOG:
-                do_config_power_action (manager, "button-hibernate", TRUE);
-                break;
         case SCREEN_BRIGHTNESS_UP_KEY:
         case SCREEN_BRIGHTNESS_DOWN_KEY:
         case KEYBOARD_BRIGHTNESS_UP_KEY:
         case KEYBOARD_BRIGHTNESS_DOWN_KEY:
         case KEYBOARD_BRIGHTNESS_TOGGLE_KEY:
-                do_brightness_action (manager, type);
-                break;
         case BATTERY_KEY:
-                do_battery_action (manager);
                 break;
         /* Note, no default so compiler catches missing keys */
         case CUSTOM_KEY:
@@ -2340,9 +2313,6 @@ start_media_keys_idle_cb (GsdMediaKeysMa
           g_hash_table_new_full (g_str_hash, g_str_equal,
                                  g_free, g_object_unref);
 
-        /* for the power plugin interface code */
-        manager->priv->power_settings = g_settings_new (SETTINGS_POWER_DIR);
-
         /* Logic from http://git.gnome.org/browse/gnome-shell/tree/js/ui/status/accessibility.js#n163 */
         manager->priv->interface_settings = g_settings_new (SETTINGS_INTERFACE_DIR);
         g_signal_connect (G_OBJECT (manager->priv->interface_settings), "changed::gtk-theme",
@@ -2443,11 +2413,6 @@ gsd_media_keys_manager_stop (GsdMediaKey
 
         g_clear_object (&priv->logind_proxy);
         g_clear_object (&priv->settings);
-        g_clear_object (&priv->power_settings);
-        g_clear_object (&priv->power_proxy);
-        g_clear_object (&priv->power_screen_proxy);
-        g_clear_object (&priv->power_keyboard_proxy);
-        g_clear_object (&priv->composite_device);
         g_clear_object (&priv->mpris_controller);
         g_clear_object (&priv->screencast_proxy);
 
@@ -2623,6 +2588,7 @@ xrandr_ready_cb (GObject             *so
         }
 }
 
+#if 0
 static void
 power_ready_cb (GObject             *source_object,
                 GAsyncResult        *res,
@@ -2667,6 +2633,7 @@ power_keyboard_ready_cb (GObject
                 g_error_free (error);
         }
 }
+#endif
 
 static void
 on_bus_gotten (GObject             *source_object,
@@ -2675,7 +2642,6 @@ on_bus_gotten (GObject             *sour
 {
         GDBusConnection *connection;
         GError *error = NULL;
-        UpClient *up_client;
 
         if (manager->priv->bus_cancellable == NULL ||
             g_cancellable_is_cancelled (manager->priv->bus_cancellable)) {
@@ -2708,40 +2674,6 @@ on_bus_gotten (GObject             *sour
                           NULL,
                           (GAsyncReadyCallback) xrandr_ready_cb,
                           manager);
-
-        g_dbus_proxy_new (manager->priv->connection,
-                          G_DBUS_PROXY_FLAGS_NONE,
-                          NULL,
-                          GSD_DBUS_NAME ".Power",
-                          GSD_DBUS_PATH "/Power",
-                          GSD_DBUS_BASE_INTERFACE ".Power",
-                          NULL,
-                          (GAsyncReadyCallback) power_ready_cb,
-                          manager);
-
-        g_dbus_proxy_new (manager->priv->connection,
-                          G_DBUS_PROXY_FLAGS_NONE,
-                          NULL,
-                          GSD_DBUS_NAME ".Power",
-                          GSD_DBUS_PATH "/Power",
-                          GSD_DBUS_BASE_INTERFACE ".Power.Screen",
-                          NULL,
-                          (GAsyncReadyCallback) power_screen_ready_cb,
-                          manager);
-
-        g_dbus_proxy_new (manager->priv->connection,
-                          G_DBUS_PROXY_FLAGS_NONE,
-                          NULL,
-                          GSD_DBUS_NAME ".Power",
-                          GSD_DBUS_PATH "/Power",
-                          GSD_DBUS_BASE_INTERFACE ".Power.Keyboard",
-                          NULL,
-                          (GAsyncReadyCallback) power_keyboard_ready_cb,
-                          manager);
-
-        up_client = up_client_new ();
-        manager->priv->composite_device = up_client_get_display_device (up_client);
-        g_object_unref (up_client);
 }
 
 static void
--- origsrc/gnome-settings-daemon-3.14.2/plugins/xrandr/gsd-xrandr-manager.c	2014-10-17 10:25:33.000000000 -0500
+++ src/gnome-settings-daemon-3.14.2/plugins/xrandr/gsd-xrandr-manager.c	2015-01-18 03:30:48.986901100 -0600
@@ -36,7 +36,6 @@
 #include <gio/gio.h>
 #include <gdk/gdk.h>
 #include <gtk/gtk.h>
-#include <libupower-glib/upower.h>
 
 #define GNOME_DESKTOP_USE_UNSTABLE_API
 
@@ -92,8 +91,6 @@ struct GsdXrandrManagerPrivate {
         GnomeRRScreen *rw_screen;
         gboolean running;
 
-        UpClient *upower_client;
-
         GSettings       *settings;
         GDBusNodeInfo   *introspection_data;
         guint            name_id;
@@ -456,7 +453,7 @@ config_is_all_off (GnomeRRConfig *config
 static gboolean
 laptop_lid_is_closed (GsdXrandrManager *manager)
 {
-        return up_client_get_lid_is_closed (manager->priv->upower_client);
+        return FALSE; //up_client_get_lid_is_closed (manager->priv->upower_client);
 }
 
 static gboolean
@@ -1217,8 +1214,6 @@ on_rr_screen_acquired (GObject      *obj
                 return;
         }
 
-        manager->priv->upower_client = up_client_new ();
-
         log_msg ("State of screen at startup:\n");
         log_screen (manager->priv->rw_screen);
 
@@ -1271,12 +1266,6 @@ gsd_xrandr_manager_stop (GsdXrandrManage
                 manager->priv->rw_screen = NULL;
         }
 
-        if (manager->priv->upower_client != NULL) {
-                g_signal_handlers_disconnect_by_data (manager->priv->upower_client, manager);
-                g_object_unref (manager->priv->upower_client);
-                manager->priv->upower_client = NULL;
-        }
-
         if (manager->priv->name_id != 0)
                 g_bus_unown_name (manager->priv->name_id);
 
