--- origsrc/gnome-settings-daemon-3.6.3/configure.ac	2012-12-19 16:55:41.314691200 -0600
+++ src/gnome-settings-daemon-3.6.3/configure.ac	2012-12-19 17:08:10.622549200 -0600
@@ -212,7 +212,7 @@ dnl ------------------------------------
 dnl - xrandr plugin stuff
 dnl ---------------------------------------------------------------------------
 
-PKG_CHECK_MODULES(XRANDR, [gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION upower-glib >= $UPOWER_REQUIRED_VERSION])
+PKG_CHECK_MODULES(XRANDR, [gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION])
 
 dnl ---------------------------------------------------------------------------
 dnl - orientation plugin stuff
@@ -231,8 +231,6 @@ PKG_CHECK_MODULES(SOUND, [libpulse >= $P
 # ---------------------------------------------------------------------------
 # Power
 # ---------------------------------------------------------------------------
-PKG_CHECK_MODULES(POWER, upower-glib >= $UPOWER_REQUIRED_VERSION gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3 libnotify x11 xext)
-
 if test x$have_gudev != xno; then
 	PKG_CHECK_MODULES(BACKLIGHT_HELPER,
 		glib-2.0 >= $GLIB_REQUIRED_VERSION
@@ -511,7 +503,6 @@ plugins/a11y-settings/Makefile
 plugins/common/Makefile
 plugins/cursor/Makefile
 plugins/dummy/Makefile
-plugins/power/Makefile
 plugins/housekeeping/Makefile
 plugins/keyboard/Makefile
 plugins/media-keys/Makefile
--- origsrc/gnome-settings-daemon-3.2.2/plugins/Makefile.am	2011-11-08 07:48:52.000000000 -0600
+++ src/gnome-settings-daemon-3.2.2/plugins/Makefile.am	2011-11-17 20:14:31.694027600 -0600
@@ -10,7 +10,6 @@ enabled_plugins =	\
 	cursor		\
 	datetime	\
 	dummy		\
-	power		\
 	housekeeping    \
 	keybindings	\
 	keyboard	\
--- origsrc/gnome-settings-daemon-3.4.1/plugins/xrandr/gsd-xrandr-manager.c	2012-04-12 04:54:14.000000000 -0500
+++ src/gnome-settings-daemon-3.4.1/plugins/xrandr/gsd-xrandr-manager.c	2012-05-06 23:35:05.586106000 -0500
@@ -38,7 +38,9 @@
 #include <gdk/gdk.h>
 #include <gdk/gdkx.h>
 #include <gtk/gtk.h>
+#if 0
 #include <libupower-glib/upower.h>
+#endif
 #include <X11/Xatom.h>
 
 #define GNOME_DESKTOP_USE_UNSTABLE_API
@@ -104,8 +106,10 @@ struct GsdXrandrManagerPrivate
         GnomeRRScreen *rw_screen;
         gboolean running;
 
+#if 0
         UpClient *upower_client;
         gboolean laptop_lid_is_closed;
+#endif
 
         GSettings       *settings;
         GDBusNodeInfo   *introspection_data;
@@ -461,8 +465,10 @@ apply_configuration_from_filename (GsdXr
                 }
         }
 
+#if 0
         if (up_client_get_lid_is_closed (priv->upower_client))
                 turn_off_laptop_display_in_configuration (priv->rw_screen, config);
+#endif
 
         gnome_rr_config_ensure_primary (config);
 	success = gnome_rr_config_apply_with_time (config, priv->rw_screen, timestamp, error);
@@ -850,7 +856,11 @@ config_is_all_off (GnomeRRConfig *config
 static gboolean
 laptop_lid_is_closed (GsdXrandrManager *manager)
 {
+#if 0
         return up_client_get_lid_is_closed (manager->priv->upower_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
@@ -1927,6 +1937,7 @@ turn_off_laptop_display (GsdXrandrManage
         g_object_unref (config);
 }
 
+#if 0
 static void
 power_client_changed_cb (UpClient *client, gpointer data)
 {
@@ -1953,6 +1964,7 @@ power_client_changed_cb (UpClient *clien
                         use_stored_configuration_or_auto_configure_outputs (manager, GDK_CURRENT_TIME);
         }
 }
+#endif
 
 gboolean
 gsd_xrandr_manager_start (GsdXrandrManager *manager,
@@ -1976,10 +1988,12 @@ gsd_xrandr_manager_start (GsdXrandrManag
 
         g_signal_connect (manager->priv->rw_screen, "changed", G_CALLBACK (on_randr_event), manager);
 
+#if 0
         manager->priv->upower_client = up_client_new ();
         manager->priv->laptop_lid_is_closed = up_client_get_lid_is_closed (manager->priv->upower_client);
         g_signal_connect (manager->priv->upower_client, "changed",
                           G_CALLBACK (power_client_changed_cb), manager);
+#endif
 
         log_msg ("State of screen at startup:\n");
         log_screen (manager->priv->rw_screen);
@@ -2025,11 +2039,13 @@ gsd_xrandr_manager_stop (GsdXrandrManage
                 manager->priv->rw_screen = NULL;
         }
 
+#if 0
         if (manager->priv->upower_client != NULL) {
                 g_signal_handlers_disconnect_by_data (manager->priv->upower_client, manager);
                 g_object_unref (manager->priv->upower_client);
                 manager->priv->upower_client = NULL;
         }
+#endif
 
         if (manager->priv->introspection_data) {
                 g_dbus_node_info_unref (manager->priv->introspection_data);
