--- origsrc/gnome-settings-daemon-3.4.1/configure.ac	2012-05-07 00:00:35.159592600 -0500
+++ src/gnome-settings-daemon-3.4.1/configure.ac	2012-05-07 00:04:34.799299200 -0500
@@ -203,7 +203,27 @@ dnl ------------------------------------
 dnl - wacom
 dnl ---------------------------------------------------------------------------
 
-PKG_CHECK_MODULES(WACOM, [libwacom >= $LIBWACOM_REQUIRED_VERSION x11 xi xtst $GUDEV_PKG gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION xorg-wacom])
+have_wacom=false
+AC_ARG_ENABLE(wacom,
+  AC_HELP_STRING([--disable-wacom],
+                 [turn off Wacom tablet support]),
+       [case "${enableval}" in
+               yes) WANT_WACOM=yes ;;
+               no)  WANT_WACOM=no ;;
+               *) AC_MSG_ERROR(bad value ${enableval} for --disable-wacom) ;;
+       esac],
+       [WANT_WACOM=yes]) dnl Default value
+
+if test x$WANT_WACOM = xyes ; then
+       PKG_CHECK_MODULES(WACOM, [libwacom >= $LIBWACOM_REQUIRED_VERSION x11 xi xtst $GUDEV_PKG gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION xorg-wacom],
+             [have_wacom=true
+              AC_DEFINE(HAVE_WACOM, 1, [Define if UPower should be used])],
+             [have_wacom=false])
+fi
+AM_CONDITIONAL(HAVE_WACOM, test "x$have_wacom" = "xtrue")
+
+AC_SUBST(WACOM_CFLAGS)
+AC_SUBST(WACOM_LIBS)
 
 dnl ==============================================
 dnl PackageKit section
--- origsrc/gnome-settings-daemon-3.4.1/plugins/Makefile.am	2012-05-06 22:39:57.014866500 -0500
+++ src/gnome-settings-daemon-3.4.1/plugins/Makefile.am	2012-05-06 22:47:31.438858100 -0500
@@ -14,7 +14,6 @@ enabled_plugins =	\
 	media-keys	\
 	mouse		\
 	sound		\
-	wacom		\
 	xrandr		\
 	xsettings	\
 	$(NULL)
@@ -51,5 +50,11 @@ else
 disabled_plugins += print-notifications
 endif
 
+if HAVE_WACOM
+enabled_plugins += wacom
+else
+disabled_plugins += wacom
+endif
+
 SUBDIRS = common $(enabled_plugins)
 DIST_SUBDIRS = $(SUBDIRS) $(disabled_plugins)
